cmake/modules/snippets.cmake

```diff
@@ -93,14 +93,11 @@ function(zephyr_process_snippets)
   endif()
   include(${snippets_generated})
 
-  # Propagate include()d build system settings to the caller.
-  set(DTC_OVERLAY_FILE ${DTC_OVERLAY_FILE} PARENT_SCOPE)
-  set(OVERLAY_CONFIG ${OVERLAY_CONFIG} PARENT_SCOPE)
-
   # Create the 'snippets' target. Each snippet is printed in a
   # separate command because build system files are not fond of
   # newlines.
-  list(TRANSFORM SNIPPET_NAMES PREPEND "COMMAND;${CMAKE_COMMAND};-E;echo;"
+  get_target_property(snippet_names snippets_target NAMES)
+  list(TRANSFORM snippet_names PREPEND "COMMAND;${CMAKE_COMMAND};-E;echo;"
     OUTPUT_VARIABLE snippets_target_cmd)
   add_custom_target(snippets ${snippets_target_cmd} USES_TERMINAL)
```

Comment:
@mbolivar-nordic I considered to have just a single `snippets` target for this and then use the same target for scoping the variables as for the print command.
But because the printing itself has been constructed so that each snippet name requires `COMMAND` in front, then we need to fetch the list prior to creating the custom target. (we cannot a generator expression in this case).

If the `snippets` target itself is just for debugging, then just printing a space separated list of snippets would make it possible to simplify the code, but before proposing so, then I would like to know if there are special requirements for the `snippets` targets so that it must use newlines.

Reply:
This is done for consistency with boards and shields. I expect the number of snippets may grow too large over time to comfortably fit on one line.

Reply:
thanks for the explanation.

keeping as-is for now, but will consider to rework into a dedicated call to `snippets.py` when `snippets` target is specifically invoked.
That will be even more consistent with what we do for boards, and remove the need for handling the list here, thus making it possible to make cleaner CMake code.